### DISCUSSION TITLE
Allow uploading JPEG XL files

### DISCUSSION THREAD

**Discussion by mo271** [community/community#18139] [Other Feature Feedback, Questions, & Ideas]
This is a feature request to allow the uploading of images with the ".jxl" file ending.

Currently it is already possible to upload JPEG XL images, but one has to add a file ending that passes the upload, for example ".jpg". This doesn't make the image a legacy JPEG image, the bytes are unchanged, and if your browser can display those files, they will be displayed. (Here's an example of a jxl image uploaded in such a way: https://github.com/libjxl/libjxl/issues/1478#issuecomment-1151053900, and here the link to the image: https://user-images.githubusercontent.com/3491627/172845157-cb12815a-e16a-4e4a-9558-1c5a793135c6.jpg)

So all that is needed is to allow the file ending in the upload dialog, everything else is already supported. 

JPEG XL is getting more and more adoption (see https://github.com/libjxl/libjxl/blob/main/doc/software_support.md), and of course many projects that adopt it have their code on github and it would therefore be convenient to be able to attach images for debugging etc. on discussions, even if the browser flags to view the images is not enabled.

**Comment by sandstrom** [comment]
## Support for JPEG-XL in Github LFS

Related to this one, it would also be good with JPEG-XL support for rendering in Git LFS (Large File Storage).

Right now, when using Git LFS via Github, we get rendered images in the diffs, for JPEG. This is great!

However, we're switching our authoring pipeline over to JPEG-XL, because it brings many advantages to our image workflows.

We'd still be using Github LFS though, and would be neat if they would show rendered in the PR preview/diffs.

**Comment by ShalokShalom** [comment]
WebP support got added recently (!), and [AVIF support](https://github.com/orgs/community/discussions/138164) is still unanswered.
